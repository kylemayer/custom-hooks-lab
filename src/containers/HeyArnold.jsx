import React from 'react';
import CharacterList from '../components/displays/CharacterList';
import { useCharacters } from '../hooks/useCharacters';

const HeyArnold = () => {
  const { loading, characters } = useCharacters();

  if (loading) return <h4>Loading...</h4>;

  return <CharacterList characters={characters} />;
};

export default HeyArnold;
