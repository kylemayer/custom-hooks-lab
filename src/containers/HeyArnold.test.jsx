/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import HeyArnold from './HeyArnold';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import list from '../json/list.json';

const server = setupServer(
  rest.get(
    'https://hey-arnold-api.herokuapp.com/api/v1/characters',
    (req, res, ctx) => {
      return res(ctx.json(list));
    }
  )
);

describe('HeyArnold Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of hey arnold characters', async () => {
    const { container } = render(
      <MemoryRouter>
        <HeyArnold />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
