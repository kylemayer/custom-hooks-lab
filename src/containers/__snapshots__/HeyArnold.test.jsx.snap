// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeyArnold Container displays a list of hey arnold characters 1`] = `
<div>
  <ul
    aria-label="characters"
  >
    <a
      href="/character/5da237699734fdcb7bef8f5c"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Nancy"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
          />
          <h4>
            Nancy
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f59"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Grandpa's grandfather"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
          />
          <h4>
            Grandpa's grandfather
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f52"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Grandpa Phil"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
          />
          <h4>
            Grandpa Phil
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f58"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Grandpa's father"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
          />
          <h4>
            Grandpa's father
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f70"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Phoebe Heyerdahl"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/93/Phoebe_%28classic%29.png/revision/latest/scale-to-width-down/247?cb=20190816022206"
          />
          <h4>
            Phoebe Heyerdahl
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f7e"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Eugene Horowitz"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/9/99/EAFDB121-EC13-4F12-849E-37037A95E3D6.png/revision/latest/scale-to-width-down/270?cb=20190816022130"
          />
          <h4>
            Eugene Horowitz
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f82"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Ray"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/3/34/Ray.png/revision/latest/scale-to-width-down/310?cb=20181006091830"
          />
          <h4>
            Ray
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f8b"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Shelley"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
          />
          <h4>
            Shelley
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f76"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Lila Sawyer"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b0/Lila_Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171212053404"
          />
          <h4>
            Lila Sawyer
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f7c"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Uncle Stinky"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Uncle_Stinky.jpg/revision/latest/scale-to-width-down/310?cb=20120924204406"
          />
          <h4>
            Uncle Stinky
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f55"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Arnie"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
          />
          <h4>
            Arnie
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f66"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Olga Pataki"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/5/5e/Olga_Pataki.png/revision/latest/scale-to-width-down/310?cb=20180124053913"
          />
          <h4>
            Olga Pataki
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f6b"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Timberly Johanssen"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/Timberly.png/revision/latest/scale-to-width-down/310?cb=20171210045406"
          />
          <h4>
            Timberly Johanssen
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f6d"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Jerry Berman"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2f/Jerry_Berman.png/revision/latest/scale-to-width-down/310?cb=20171219061306"
          />
          <h4>
            Jerry Berman
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f73"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Rhonda Wellington Lloyd"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/b4/C1BBB448-FE38-4F7B-98FA-2137F277F9C5.png/revision/latest/scale-to-width-down/213?cb=20190816022649"
          />
          <h4>
            Rhonda Wellington Lloyd
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f78"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Stinky Peterson"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/2/24/Stinky_Peterson.png/revision/latest/scale-to-width-down/310?cb=20160516072820"
          />
          <h4>
            Stinky Peterson
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f71"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Kyo Heyerdahl"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cf/Kyo_Heyerdahl.png/revision/latest/scale-to-width-down/310?cb=20190310221807"
          />
          <h4>
            Kyo Heyerdahl
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f8e"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Iggy"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/0/0e/Iggy.png/revision/latest/scale-to-width-down/310?cb=20190210045957"
          />
          <h4>
            Iggy
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f77"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Mr. Sawyer"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/4/45/Mr._Sawyer.png/revision/latest/scale-to-width-down/310?cb=20171219055204"
          />
          <h4>
            Mr. Sawyer
          </h4>
        </div>
      </li>
    </a>
    <a
      href="/character/5da237699734fdcb7bef8f84"
    >
      <li
        class="list"
      >
        <div>
          <img
            alt="Thaddeus \\"Curly\\" Gammelthorpe"
            src="https://vignette.wikia.nocookie.net/heyarnold/images/b/ba/Classic_Curly_promo.png/revision/latest/scale-to-width-down/246?cb=20181010011614"
          />
          <h4>
            Thaddeus "Curly" Gammelthorpe
          </h4>
        </div>
      </li>
    </a>
  </ul>
</div>
`;
